src/simulators/matrix_product_state/matrix_product_state.hpp

```diff
@@ -140,6 +143,12 @@ class State : public Base::State<matrixproductstate_t> {
 
   virtual void add_metadata(ExperimentResult &result) const override;
 
+  // Similar to add_metadata, but used to add data at the end of circuit execution
+  virtual void add_reporting_metadata(ExperimentResult &result) const override;
```

Comment:
I think we can remove this and just use add_metadata, but update the controller code to only add metadata at the end of simulation, rather than at the start.

Reply:
done, see above.